internal/api/installation_test.go

```diff
@@ -779,21 +783,49 @@ func TestLeaveGroup(t *testing.T) {
 			require.True(t, unlocked)
 		}()
 
-		err = client.LeaveGroup(installation1.ID)
+		err = client.LeaveGroup(installation1.ID, &model.LeaveGroupRequest{RetainConfig: true})
 		require.EqualError(t, err, "failed with status code 409")
 	})
 
 	t.Run("while in group 1", func(t *testing.T) {
-		err = client.LeaveGroup(installation1.ID)
-		require.NoError(t, err)
+		t.Run("don't retain group config", func(t *testing.T) {
```

Comment:
What is the outcome of a test where an Installation has configuration A and is joined to a group with configuration B and then leaves the group with `retain-config == true`, as opposed to this test which sets these values once the Installation is already part of the Group?

Reply:
Installations are not directly updated when they join a group so their original config remains. When leaving the group, you have the option of updating the installation to the group-merged version as it leaves (resulting in a no-op basically) or just leaving without the merge which could result in it updating its behavior.

The tests here try to cover both options. I think that answered your question?

Reply:
Maybe I'm wrong but the test as-written doesn't seem like it actually tests leaving without the merge because the data is written to the Installation after it's joined the group. 

My concern is that when the Installation is joined to the group, the configuration held by the Installation is overwritten by the Group's config (the merged config). Then, when the Installation is moved out of the Group, we "preserve" the Installation config, but it was already overwritten on Join.

Reply:
After reading your answer here https://github.com/mattermost/mattermost-cloud/pull/186#discussion_r401795700 and thinking a bit more, I think that this test is _probably_ fine as-is but I'll leave my concern above just in case I did spot something you think is worthwhile.

Reply:
Oh, there are plenty of other tests in various places that check installation config merge behavior when joined to groups if that is the primary concern.

This test is in the leave group test suite and so I was just focusing on that flow working correctly.